pkg/manager/overrides.go

```diff
@@ -0,0 +1,82 @@
+package manager
+
+import natsv1alpha1 "github.com/kyma-project/nats-manager/api/v1alpha1"
+
+const (
+	MinClusterSize         = 3
+	IstioEnabledKey        = "istio.enabled"
+	RotatePasswordKey      = "auth.rotatePassword" //nolint:gosec // Its is not password.
+	ClusterEnabledKey      = "cluster.enabled"
+	ClusterSizeKey         = "cluster.replicas"
+	FileStorageClassKey    = "nats.jetstream.fileStorage.storageClassName"
+	FileStorageSizeKey     = "global.jetstream.fileStorage.size"
+	MemStorageEnabledKey   = "nats.jetstream.memStorage.enabled"
+	MemStorageSizeKey      = "nats.jetstream.memStorage.size"
+	DebugEnabledKey        = "nats.logging.debug"
+	TraceEnabledKey        = "nats.logging.trace"
+	CommonLabelsKey        = "commonLabels"
+	CommonAnnotationsKey   = "commonAnnotations"
+	ResourceRequestsCPUKey = "nats.resources.requests.cpu"
+	ResourceRequestsMemKey = "nats.resources.requests.memory"
+	ResourceLimitsCPUKey   = "nats.resources.limits.cpu"
+	ResourceLimitsMemKey   = "nats.resources.limits.memory"
+)
+
+func (m NATSManager) GenerateOverrides(spec *natsv1alpha1.NATSSpec, istioEnabled bool,
+	rotatePassword bool) map[string]interface{} {
+	overrides := map[string]interface{}{
+		IstioEnabledKey:   istioEnabled,
+		RotatePasswordKey: rotatePassword,
+	}
+
+	// clustering
+	overrides[ClusterSizeKey] = spec.Cluster.Size
+	overrides[ClusterEnabledKey] = true
+	if spec.Cluster.Size < MinClusterSize {
```

Comment:
i assume we will check this later using validation rules in the api-server?!

Reply:
yes, validation and defaulting will be done in the api-server part.